code/_onclick/hud/hud.dm

```diff
@@ -205,6 +205,83 @@ datum/hud/New(mob/owner)
 	else if(isguardian(mymob))
 		guardian_hud()
 
+	create_parallax()
+
+
+/client/var/list/obj/screen/spessbg/spessbg = list()
+
+var/list/client/parallax_on_clients = list()
+
+var/area/global_space_area = null
+
+/obj/screen/spessbg
+	var/offset_x = 0
+	var/offset_y = 0
+/datum/hud/var/showing_parallax
+/datum/hud/proc/create_parallax()
+	var/client/C = mymob.client
+	showing_parallax = C.prefs.space_parallax
+	if (C.prefs.space_parallax)
+		parallax_on_clients |= C
+	else
+		parallax_on_clients -= C
+	for(var/area/A in all_areas)
+		if(A.white_overlay)
+			if (C.prefs.space_parallax)
+				C.images |= A.white_overlay
+			else
+				C.images -= A.white_overlay
+	if (C.spessbg.len)
+		for(var/obj/screen/spessbg/bgobj in C.spessbg)
+			bgobj.layer = (C.prefs.space_parallax) ? AREA_LAYER + 0.5 : 0
+			C.screen |= bgobj
+		return
+	for(var/i=0;i<4;i++)
+		var/obj/screen/spessbg/bgobj = new /obj/screen/spessbg()
+		if(i & 1)
+			bgobj.offset_x = 480
+		if(i & 2)
+			bgobj.offset_y = 480
+		bgobj.icon = 'icons/mob/screen1_full.dmi'
+		bgobj.icon_state = "spess"
+		bgobj.name = "spess"
+		bgobj.screen_loc = "CENTER-7:[bgobj.offset_x],CENTER-7:[bgobj.offset_y]"
+		bgobj.layer = (C.prefs.space_parallax) ? AREA_LAYER + 0.5 : 0
+		bgobj.blend_mode = BLEND_MULTIPLY
+		bgobj.mouse_opacity = 0
+		C.spessbg += bgobj
+		C.screen += bgobj
+	update_parallax()
+/datum/hud/proc/update_parallax()
+	var/atom/posobj = mymob
+	while(!istype(posobj,/turf))
+		posobj = posobj.loc
+	for(var/obj/screen/spessbg/bgobj in mymob.client.spessbg)
+		bgobj.screen_loc = "CENTER-7:[bgobj.offset_x-posobj.x],CENTER-7:[bgobj.offset_y-posobj.y]"
+
+/*///hook_handler/parallax/proc/OnMobAreaChange(var/list/args)
```

Comment:
Wow, I completely forgot about that. That was a failed attempt at fixing something.